### PR TITLE
[WIP][Feautre, Hardware] add initial suport for ascend npu

### PR DESCRIPTION
**Pull request by 22dimensions**
## Motivation
see issue https://github.com/sgl-project/sglang/issues/3781

## Modifications

1. add new device args 'npu'
2. import torch_npu and adapt torch related code
3. add usage document

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

## Examples

```bash
python -m sglang.launch_server --model-path LLM-Research/Llama-3.2-1B-Instruct  --device npu --grammar-backend outlines --attention-backend torch_native  --port 30000 --host 0.0.0.0
```

image

### CODE REVIEW

**Comment by 22dimensions** [comment]
https://github.com/sgl-project/sglang/pull/3853 is merged, so i close this pr.
